controllers/cloud.redhat.com/providers/pullsecrets/default.go

```diff
@@ -0,0 +1,167 @@
+package pullsecrets
+
+import (
+	"fmt"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/config"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/providers"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/providers/serviceaccount"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/utils"
+
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type pullsecretProvider struct {
+	providers.Provider
+}
+
+// CoreConfigSecret is the config that is presented as the cdappconfig.json file.
+var CoreConfigSecret = providers.NewSingleResourceIdent(ProvName, "core_config_secret", &core.Secret{})
+
+// NewPullSecretProvider returns a new End provider run at the end of the provider set.
+func NewPullSecretProvider(p *providers.Provider) (providers.ClowderProvider, error) {
+
+	appList := &crd.ClowdAppList{}
+	if err := p.Env.GetAppsInEnv(p.Ctx, p.Client, appList); err != nil {
+		return nil, err
+	}
+
+	namespaceList := map[string]bool{}
+
+	for _, app := range appList.Items {
+		namespaceList[app.Namespace] = true
+	}
+
+	secList, err := copyPullSecrets(p, namespaceList)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sa := &core.ServiceAccount{}
+	if err := p.Cache.Get(serviceaccount.CoreEnvServiceAccount, sa); err != nil {
+		return nil, err
+	}
+
+	addAllSecrets(secList, sa)
+
+	if err := p.Cache.Update(serviceaccount.CoreEnvServiceAccount, sa); err != nil {
+		return nil, err
+	}
+
+	return &pullsecretProvider{Provider: *p}, nil
+}
+
+func copyPullSecrets(prov *providers.Provider, namespaceList map[string]bool) ([]string, error) {
+	var secList []string
+
+	for _, pullSecretName := range prov.Env.Spec.Providers.PullSecrets {
+		for namespace := range namespaceList {
+			if namespace == pullSecretName.Namespace {
```

Comment:
let's not make an exception for this.  just copy it to a new secret in the same namespace if applicable